plugins/woocommerce/src/Admin/PluginsInstallLoggers/AsyncPluginsInstallLogger.php

```diff
@@ -135,14 +135,97 @@ public function add_error( string $plugin_name, string $error_message = null ) {
 	/**
 	 * Record completed_time.
 	 *
+	 * @param array $data return data from install_plugins().
 	 * @return void
 	 */
-	public function complete() {
+	public function complete( $data = array() ) {
 		$option = $this->get();
 
 		$option['complete_time'] = time();
 		$option['status']        = 'complete';
 
+		$this->track( $data );
 		$this->update( $option );
 	}
+
+	private function get_plugin_track_key( $id ) {
+		$slug = explode( ':', $id )[0];
+		$key  = preg_match( '/^woocommerce(-|_)payments$/', $slug )
+			? 'wcpay'
+			: explode( ':', str_replace( '-', '_', $slug ) )[0];
+		return $key;
+	}
+
+	/**
+	 * Returns time frame for a given time in milliseconds.
+	 *
+	 * @param int $timeInMs - time in milliseconds
+	 *
+	 * @return string - Time frame.
+	 */
+	function get_timeframe( $timeInMs ) {
+		$timeFrames = [
+			[
+				'name' => '0-2s',
+				'max'  => 2,
+			],
+			[
+				'name' => '2-5s',
+				'max'  => 5,
+			],
+			[
+				'name' => '5-10s',
+				'max'  => 10,
+			],
+			[
+				'name' => '10-15s',
+				'max'  => 15,
+			],
+			[
+				'name' => '15-20s',
+				'max'  => 20,
+			],
+			[
+				'name' => '20-30s',
+				'max'  => 30,
+			],
+			[
+				'name' => '30-60s',
+				'max'  => 60,
+			],
+			[ 'name' => '>60s' ],
+		];
+
+		foreach ( $timeFrames as $timeFrame ) {
```

Comment:
✍🏼 Here and elsewhere we are breaking the `WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase` linting rule (ie, prefer snake_case). Unfortunately, we can't currently rely on the PHPCS checks that run via GitHub Actions (they are reporting success when they haven't actually run).

Reply:
Thank you for catching it 👍 I converted JS version of code to PHP and followed variable name convention from JS 🤦 I'll make a fix.

Reply:
Fixed in https://github.com/woocommerce/woocommerce/pull/39921/commits/178e6eeb796e0b737db5b7db1ea79bac45bd5b19